### PR TITLE
Auto Testing

### PR DESCRIPTION
**Pull request by kaijchang**
Merging bare-bones trajectory code so we can add logging and other stuff onto it.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 language: java
 
 script:
-  - ./gradlew test
\ No newline at end of file
+  - ./gradlew test
+
+os:
+  - osx
```
Comment (ri-nath) [inline .travis.yml:7]: what does this do?

File: src/main/java/org/team4159/frc/robot/Trajectories.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  );
+
+  private static TrajectoryConfig config = new TrajectoryConfig(
+    DRIVE_CONSTANTS.MAX_TRAJECTORY_SPEED,
+    DRIVE_CONSTANTS.MAX_TRAJECTORY_ACCELERATION
+  )
+    .setKinematics(kinematics)
+    .addConstraint(constraint);
+
+  public static final Trajectory GO_FORWARD_ONE_METER =
```
Comment (ri-nath) [inline src/main/java/org/team4159/frc/robot/Trajectories.java]: probably should rename this

File: src/main/java/org/team4159/frc/robot/commands/auto/FollowTrajectory.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      new ChassisSpeeds(initialState.velocityMetersPerSecond,
+        0,
+        initialState.curvatureRadPerMeter
+          * initialState.velocityMetersPerSecond));
+    timer.reset();
+    timer.start();
+  }
+
+  @Override
+  public void execute() {
```
Comment (ri-nath) [inline src/main/java/org/team4159/frc/robot/commands/auto/FollowTrajectory.java:61]: maybe just migrate to RamseteCommand? (unless we want to keep for smart dashboard stuff)

File: src/test/java/org/team4159/frc/robot/TestTrajectories.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+import edu.wpi.first.wpilibj.trajectory.Trajectory;
+import org.team4159.lib.logging.CSVWriter;
+
+public class TestTrajectories {
+  @Test
+  public void TestTrajectoryGeneration() {
+    Field[] fields = Trajectories.class.getFields();
```
Comment (ri-nath) [inline src/test/java/org/team4159/frc/robot/TestTrajectories.java:15]: this is cool

**Review by ri-nath — CHANGES_REQUESTED** [review]
check comments

File: src/main/java/org/team4159/frc/robot/commands/auto/FollowTrajectory.java (hunk 3 above)
Comment (kaijchang) [reply src/main/java/org/team4159/frc/robot/commands/auto/FollowTrajectory.java:61]: going to add logging stuff eventually

File: .travis.yml (hunk 1 above)
Comment (kaijchang) [reply .travis.yml:7]: there's no desktop library for wpilib on linux so i have to tell it to run the build on osx

File: src/main/java/org/team4159/frc/robot/Trajectories.java (hunk 2 above)
Comment (kaijchang) [reply src/main/java/org/team4159/frc/robot/Trajectories.java]: maybe

**Review by ri-nath — APPROVED** [review]

**Pull request merged**